### PR TITLE
Adding laundry-dryer-controls cluster XML

### PR DESCRIPTION
**Pull request by eahove**
Created XML for the laundry-dryer-controls cluster and related generated code. For 0.7 ballot.

This is based on the updated spec from PR:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7766

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -40,15 +41,18 @@ jobs:
         env:
             BUILD_VARIANT: ${{matrix.build_variant}}
             CHIP_TOOL_VARIANT: ${{matrix.chip_tool}}
-            TSAN_OPTIONS: "halt_on_error=1 suppressions=scripts/tests/chiptest/tsan-linux-suppressions.txt"
+            TSAN_OPTIONS:
```
Comment (bzbarsky-apple) [inline .github/workflows/tests.yaml:44]: Whatever auto-formatter is being used here disagrees with previous auto-formatters.  Please undo all the unrelated changes.

File: src/app/zap-templates/zcl/data-model/chip/laundry-dryer-controls-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<configurator>
+    <domain name="CHIP" />
+
+    <enum name="DrynessLevelEnum" type="enum8">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/laundry-dryer-controls-cluster.xml:21]: This is not what the spec has, as far as I can tell.  What exact spec version is this targeting, if it's not master or 0.7 ballot?  That needs to be clearly stated in the PR for this to be reviewable.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/laundry-dryer-controls-cluster.xml (hunk 2 above)
Comment (eahove) [reply src/app/zap-templates/zcl/data-model/chip/laundry-dryer-controls-cluster.xml:21]: This is based on an updated version of the spec that is still in PR:
https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/7766
